Source/WebCore/workers/service/server/SWServer.cpp

```diff
@@ -304,20 +313,7 @@ void SWServer::clearAll(CompletionHandler<void()>&& completionHandler)
     m_registrationStore->clearAll(WTFMove(completionHandler));
 }
 
-void SWServer::startSuspension(CompletionHandler<void()>&& completionHandler)
-{
-    if (!m_registrationStore) {
-        completionHandler();
-        return;
-    }
-    m_registrationStore->startSuspension(WTFMove(completionHandler));
-}
-
-void SWServer::endSuspension()
-{
-    if (m_registrationStore)
-        m_registrationStore->endSuspension();
-}
+// TODO: suspend registration database via NetworkStorageManager
```

Comment:
What adoption do we need here?
Please file a bug if we need to do anything.

Reply:
Oh I forgot to drop this. The storage queue of `NetworkStorageManager`, which `ServiceWorkerStorageManager` runs on will stop processing task when `NetworkStorageManager::suspend`, so there is special handling needed for `ServiceWorkerStorageManager` (because `SWRegistrationDatabase` does not create transaction that outlives WorkQueue task).